prow/plugins/lifecycle/lifecycle.go

```diff
@@ -121,10 +106,34 @@ func handleOne(gc lifecycleClient, log *logrus.Entry, e *github.GenericCommentEv
 	remove := mat[1] != ""
 	cmd := mat[2]
 	lbl := "lifecycle/" + cmd
-	// Let's start simple and allow anyone to add/remove frozen, stale, putrid, rotten labels.
+
+	// Let's start simple and allow anyone to add/remove frozen, stale, rotten labels.
 	// Adjust if we find evidence of the community abusing these labels.
-	if remove {
+	labels, err := gc.GetIssueLabels(org, repo, number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get labels.")
+	}
+
+	// If the label exists and we asked for it to be removed, remove it.
+	if github.HasLabel(lbl, labels) && remove {
 		return gc.RemoveLabel(org, repo, number, lbl)
 	}
-	return gc.AddLabel(org, repo, number, lbl)
+
+	// If the label does not exist and we asked for it to be added,
+	// remove other existing lifecycle labels and add it.
+	if !github.HasLabel(lbl, labels) && !remove {
+		for _, label := range lifecycleLabels {
+			if label != lbl && github.HasLabel(label, labels) {
+				if err := gc.RemoveLabel(org, repo, number, label); err != nil {
+					log.WithError(err).Errorf("Github failed to remove the following label: %s", label)
+				}
+			}
+		}
+
+		if err := gc.AddLabel(org, repo, number, lbl); err != nil {
```

Comment:
If removing some of the labels resulted in an error, do we still want to add labels?

Don't have a strong opinion here, but, just note that technically there's no guarantee we don't wind up with multiple lifecycle labels (there's also a race between GetIssueLabels and the RemoveLabel/AddLabel)